lib/components/lenra_checkbox.dart

```diff
@@ -65,10 +72,21 @@ class LenraApplicationCheckbox extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    if(name != null) {
+      context.read<FormProvider>().setFormFieldValue(name!, value);
+    }
+
     return LenraCheckbox(
       value: value,
       tristate: tristate ?? false,
-      onPressed: onPressed == null ? null : (v) => onCheck(context, v),
+      onPressed: onPressed == null
```

Comment:
Two thing on this : 
- First, a ternary expression should only be used for simple one-liner. Otherwise it's too hard to read.
- Secondly, here, the form will not work if we are not adding the "onPressed" property. This is not the correct behavior IMO.

It could be something like this
```dart
void handleOnPressed(context, v) {
   if (name != null) {
    FormProvider? formProvider = context.read<?FormProvider>();
      if(formProvider != null) {
        formProvider.setFormFieldValue(!name, v)
      }
    }
    if(onPressed) onCheck(context, v);
}
// [...] 

onPressed: (v) => handleOnPressed(context, v),
```


Note : this review is also the same for all other form inputs !

Reply:
The problem is that the current value of the component is changed thanks to the onPressed property by the app developer.

Setting the 
```dart
if (name != null) {
  context.read<FormProvider>().setFormFieldValue(name!, v);
}
```
code outside of the onPressed==null check will result in the component not changing its state UI-wise, but changing its state form-wise